### PR TITLE
devguide/transactions: refine some explanations. - Tx documentation v11

### PR DESCRIPTION
**Pull request by jufajardini**
Describe changes: 
- replace frontend/backend with client/server
- refine wording in Progress Tracking section
- explain better the DoS by quadratic complexity

Link to Redmine issue: https://redmine.openinfosecfoundation.org/issues/4396

Previous PR: https://github.com/OISF/suricata/pull/6265

Current state with images looks like this: https://devguide-transactions.surge.sh/extending/app-layer/transactions.html

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
Looks like a welcome documentation improvement to me

**Comment by suricata-qa** [comment]
Information:
| field | test | baseline | % |
|--|--|--|--|
| | tlpr1_stats_chk |

File: doc/devguide/Makefile.sphinx [hunk 1]
```diff
@@ -18,6 +18,7 @@ PAPEROPT_letter = -D latex_paper_size=letter
 ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
 # the i18n builder cannot share the environment and doctrees with the others
 I18NSPHINXOPTS  = $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
+BUILD_IMG = ./extending/app-layer/img/generate-images.sh
```
Comment (jasonish) [inline doc/devguide/Makefile.sphinx:21]: I think a better location for the script might be `devguide/tools`

File: doc/devguide/extending/app-layer/img/DnsRequestUnidirectionalTransaction.msc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    # Chart Options
+    arcgradient = "10";
+
+    # Entities
+    a [ label = "Client" ], b [ label = "Server" ];
+
+    # Message Flow
+    a =>> b [ label = "DNS Request" ];
+    --- [ label = "Transaction Completed" ];
+}
```
Comment (jasonish) [inline doc/devguide/extending/app-layer/img/DnsRequestUnidirectionalTransaction.msc:13]: I'd go one more step and add the DNS Response, transaction complete.  Just to complete the life cycle of a DNS request for an end-user.

File: doc/devguide/extending/app-layer/transactions.rst [hunk 3]
```diff
@@ -0,0 +1,310 @@
+************
+Transactions
+************
+
+.. contents:: Table of Contents
+
+General Concepts
+================
+
+For Suricata, transactions are abstractions that help with detecting and logging. An example of complete transaction is
```
Comment (jasonish) [inline doc/devguide/extending/app-layer/transactions.rst:10]: transactions are an abstraction

File: doc/devguide/extending/app-layer/transactions.rst [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+a pair of messages Request (from client to server) and Response (from server to client) in HTTP.
```
Comment (jasonish) [inline doc/devguide/extending/app-layer/transactions.rst:11]: a pair of messages in the form of a request (from client to server) and a response (from server to client) in HTTP.

File: doc/devguide/extending/app-layer/transactions.rst [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+In order to know when to log an event for a given protocol, the engine tracks the progress of each transaction - that
+is, when is it complete, or when does it reach a key intermediary stage. They aid during the detection phase,
```
Comment (jasonish) [inline doc/devguide/extending/app-layer/transactions.rst:14]: > or when does it reach a key intermediary stage

or when it reaches a key intermediate state

File: doc/devguide/extending/app-layer/transactions.rst [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+when dealing with protocols that can have large PDUs, like TCP, in controlling state for partial rule matching -- in case of rules that mention more than one field.
```
Comment (jasonish) [inline doc/devguide/extending/app-layer/transactions.rst:15]: Might be worth expanding PDU on first use in this chapter:
```
can have large PDUs (protocol data units)
```

**Review by jasonish — CHANGES_REQUESTED** [review]
Some grammar changes requested.  But otherwise this is looking real good.  I think some squashing of the commits will be required before merge to master since this section is all new.  But otherwise the sooner we can get this in the better.

**Comment by jufajardini** [comment]
replaced by: https://github.com/OISF/suricata/pull/6278

**Pull request closed without merging**